source/world_builder/features/continental_plate_models/grains/random_uniform_distribution_deflected.cc

```diff
@@ -0,0 +1,240 @@
+                            "The depth in meters from which the composition of this feature is present.");
+          prm.declare_entry("max depth", Types::OneOf(Types::Double(std::numeric_limits<double>::max()),Types::Array(Types::ValueAtPoints(std::numeric_limits<double>::max()))),
+                            "The depth in meters to which the composition of this feature is present.");
+
+          prm.declare_entry("compositions", Types::Array(Types::UnsignedInt(),0),
+                            "A list with the integer labels of the composition which are present there.");
+
+          prm.declare_entry("orientation operation", Types::String("replace", std::vector<std::string> {"replace"}),
+                            "Whether the value should replace any value previously defined at this location (replace) or "
+                            "add the value to the previously define value (add, not implemented). Replacing implies that all values not "
+                            "explicitly defined are set to zero.");
+
+          prm.declare_entry("grain sizes",
+                            Types::Array(Types::Double(1),0),
+                            "A list of the size of all of the grains in each composition. If set to <0, the size will be randomized between 0 and 1.");
+
+          prm.declare_entry("normalize grain sizes",
+                            Types::Array(Types::Bool(true),0),
+                            "A list of whether the sizes of the grains should be normalized or not. If normalized, the total of the grains of a composition will be equal to 1.");
+
+          prm.declare_entry("deflections",
+                            Types::Array(Types::Double(1),0),
+                            "A list of the deflection of all of the grains in each composition between 0 and 1.");
+
+          prm.declare_entry("random number seed",
+                            Types::UnsignedInt(1),
+                            "A list of the random number seed in each composition for generating random textures.");
+
+
+        }
+
+        void
+        RandomUniformDistributionDeflected::parse_entries(Parameters &prm, const std::vector<Point<2>> &coordinates)
+        {
+          min_depth_surface = Objects::Surface(prm.get("min depth",coordinates));
+          min_depth = min_depth_surface.minimum;
+          max_depth_surface = Objects::Surface(prm.get("max depth",coordinates));
+          max_depth = max_depth_surface.maximum;
+          compositions = prm.get_vector<unsigned int>("compositions");
+
+          operation = prm.get<std::string>("orientation operation");
+          grain_sizes = prm.get_vector<double>("grain sizes");
+          normalize_grain_sizes = prm.get_vector<bool>("normalize grain sizes");
+          deflections = prm.get_vector<double>("deflections");
+          seed = prm.get<unsigned int>("random number seed");
+          world->get_random_number_engine().seed(seed);
```

Comment:
As discussed, use global one

Reply:
Deleted this and other use of local seed inside features